### PR TITLE
[-Wsign-compare] warning fixes batch 3

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac 

associated_warning_ids: [
62, 64, 66, 67, 
68, 69, 71, 73,
74, 75, 76, 77,
78 
]

### CODE REVIEW

File: indexed_warning_files.json [hunk 1]
```diff
@@ -0,0 +1 @@
...platform/default/logging.cc", "508": "tensorflow/core/platform/protobuf.cc", "509": "tensorflow/core/framework/allocator_registry.cc", "510": "tensorflow/core/framework/cpu_allocator_impl.cc", "511": "tensorflow/core/kernels/lookup_util.cc", "512": "tensorflow/lite/delegates/nnapi/nnapi_delegate.cc", "513": "tensorflow/core/framework/allocator_registry.cc", "514": "tensorflow/core/common_runtime/lower_case_op.cc", "515": "tensorflow/core/data/service/compression_utils.cc", "516": "tensorflow/core/kernels/data/prefetch_autotuner.cc", "517": "tensorflow/lite/tools/versioning/runtime_version.cc", "518": "tensorflow/compiler/xla/service/hlo_evaluator.cc", "519": "tensorflow/compiler/tf2xla/xla_resource.cc", "520": "tensorflow/core/kernels/stack.cc", "521": "tensorflow/core/profiler/internal/parse_annotation.cc", "522": "tensorflow/core/framework/cpu_allocator_impl.cc", "523": "tensorflow/lite/experimental/microfrontend/lib/fft.cc", "524": "tensorflow/lite/toco/graph_transformations/ensure_uint8_weights_safe_for_fast_int8_kernels.cc", "525": "tensorflow/compiler/xla/service/while_loop_invariant_code_motion.cc", "526": "tensorflow/core/tpu/tpu_embedding_optimization_parameters_utils.cc"}
```
Comment (mihaimaruseac) [inline indexed_warning_files.json:1]: I think you don't want this file included in the PR

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 2]
```diff
@@ -538,10 +538,11 @@ bool Transposer::IsFaninPortDimsNIfConst(const utils::MutableNodeView& node,
     if (!tensor.FromProto(value_attr->tensor())) {
       return false;
     }
-    if (tensor.dims() != dims.size()) {
+    int dims_size = dims.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:541]:
```suggestion
    const int dims_size = dims.size(); 
```

Let's try to make these const if they don't change to enable a few compiler optimizations.

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 3]
```diff
@@ -863,12 +864,13 @@ inline bool IsValidConstPermTransposeNode(const utils::MutableNodeView& node,
   if (!GetValueAttrFromConstInputNode(node, IsTranspose, 1, &tensor)) {
     return false;
   }
-  if (tensor.NumElements() != permutation.size()) {
+  int permutation_size = permutation.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:867]:
```suggestion
  const int permutation_size = permutation.size();
```

File: tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc [hunk 4]
```diff
@@ -1229,10 +1231,11 @@ bool ReduceTransposer::KeepDims(const utils::MutableNodeView& node) {
 
 bool ReduceTransposer::IsAlongAxis(const Tensor& tensor,
                                    absl::Span<const int> axis, int rank) {
-  if (tensor.dims() != 1 || tensor.dim_size(0) != axis.size()) {
+  int axis_size = axis.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc:1234]:
```suggestion
  const int axis_size = axis.size();
```

File: tensorflow/core/grappler/optimizers/shape_optimizer.cc [hunk 5]
```diff
@@ -99,7 +99,8 @@ Status ShapeOptimizer::Optimize(Cluster* cluster, const GrapplerItem& item,
         }
         const auto& prop =
             properties.GetOutputProperties(reduce_indices.node->name());
-        if (prop.size() <= reduce_indices.port_id) {
+        int prop_size = prop.size();
```
Comment (mihaimaruseac) [inline tensorflow/core/grappler/optimizers/shape_optimizer.cc:102]:
```suggestion
        const int prop_size = prop.size();
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: indexed_warning_files.json (hunk 1 above)
Comment (mihaimaruseac) [inline indexed_warning_files.json:1]: You might want to add this file to `.git/config/ignore` or use `git add ..` instead of `git commit -a ...`

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by tg-at-google** [comment]
A little strange @mihaimaruseac ,  I deleted it the first time. Nothing changed. I delete it now, and it the changes reflect. No problem though, the file is gone. 👍

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**